invokeai/app/invocations/image.py

```diff
@@ -650,3 +614,134 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
             width=image_dto.width,
             height=image_dto.height,
         )
+
+class ImageNSFWBlurInvocation(BaseInvocation, PILInvocationConfig):
+    """Add blur to NSFW-flagged images"""
+    DEFAULT_ENABLED = InvokeAIAppConfig.get_config().nsfw_checker
+
+    # fmt: off
+    type: Literal["img_nsfw"] = "img_nsfw"
+
+    # Inputs
+    image: Optional[ImageField]  = Field(default=None, description="The image to check")
+    enabled: bool = Field(default=DEFAULT_ENABLED, description="Whether the NSFW checker is enabled")
```

Comment:
Agree and I wanted to do it this way, but figuring out how to create new settings in the frontend and building the graph differently depending on the setting was pretty daunting for me. Can you do the frontend work or at least give me a guide to what files need to be changed?

I'd like for this PR to go in soon because right now 3.0 users have no way to enable NSFW blurring on a shared server.

Reply:
I can do the frontend work, no worries. 

Ideally we can update the config route to report installation status of these two new models in this PR. 

Is there any easy way to check the install status? I'd rather fudge this with a hack than change a node schema between releases. 

Either way it's not a big deal and I can get the frontend updated tomorrow. Also happy to update the route, if you can point me in the right direction for checking installation status. Can work from this pr.

Reply:
I'm going to update the config route now.